### ISSUE TITLE
specify domain

### ISSUE THREAD

**Issue by austinnichols101** [thycotic/python-tss-sdk#5]
I am running into an issue where I'm getting an HTTP 400 response using python-tss-sdk against an on-premise thycotic server.  I believe the problem is that the client is not specifying the `domain`.  

Is there a way to specify the `domain`?

tagging @amigus as I'm not sure how often this repo is monitored.

**Comment by tcarolanthycotic** [comment]
@austinnichols101 - Have you tried using the following format ```domain\username``` in your username variable?

**Comment by austinnichols101** [comment]
@tcarolanthycotic 

Yes - I did try `domain\username`

I used a debug proxy (Fiddler Everwhere) and was able to determine that the x-www-form-urlencoded body needs to look like this:

```
grant_type=password&username=<username>&password=<password>&domain=<domain>
```

**Comment by Lobout** [comment]
> @tcarolanthycotic
> 
> Yes - I did try `domain\username`
> 
> I used a debug proxy (Fiddler Everwhere) and was able to determine that the x-www-form-urlencoded body needs to look like this:
> 
> ```
> grant_type=password&username=<username>&password=<password>&domain=<domain>
> ```

@austinnichols101,  you are correct if you are using a domain account it looks for a domain=<domain> value

**Comment by wsmelton** [comment]
Our REST API does not support providing a domain account with the OAuth2 token endpoint.

**Comment by briancline** [comment]
A separate `domain` field appears in one of the official scripting examples in the documentation here:
https://docs.thycotic.com/ss/10.9.0/api-scripting/rest-api-powershell-examples/index.md#managing_folders__basic_

Just doesn't appear in the authentication docs for some reason (same situation for #3).

**Comment by tcarolanthycotic** [comment]
@austinnichols101 Is absolutely right here - the standard syntax for domain username entry `domain\username` is not handled correctly by the parameter passing at the moment. This probably needs to be fixed fully, but in the meantime I created a minor pull request that shows you how you can incorporate some domain handling in to `server.py`. I've also added a working example script in `ss.py`:
#11

**Comment by amigus** [comment]
This will be fixed once #19 is merged.
